src/blister.c

```diff
@@ -1117,11 +1113,10 @@ Obj FuncLIST_BLIST (
     RequireSmallList("ListBlist", list);
     /* get and check the second argument                                   */
     RequireBlist("ListBlist", blist, "blist");
-    while ( LEN_LIST( list ) != LEN_BLIST( blist ) ) {
-        blist = ErrorReturnObj(
+    if (LEN_LIST(list) != LEN_BLIST(blist)) {
+        ErrorMayQuit(
             "ListBlist: <blist> must have the same length as <list> (%d)",
```

Comment:
You could also use `CheckSameLength` here (and elsewhere). The main challenge is that it uses `ErrorQuit` right now, not `ErrorMayQuit`. But I think changing `CheckSameLength` to use the latter would actually be an improvement. It is only u sed in 11 places. All of them are covered by tests, and relatively easy to trigger manually, so I'd be confident in making that change.

Reply:
I like it! Changing.